### ISSUE TITLE
Deploy and going live with a subset of remote players

### ISSUE THREAD

**Issue by blreams** [blreams/collegefootballpick10#88]
I wanted to start a discussion about this because I'm not sure that I grasp how this is going to work. As I mentioned in email, my plan this year is to run the pool as in the past using spreadsheets, but I would like to see if we could transition the remote players to a deployed web app on pythonanywhere. This has some implications and I want to talk through the steps needed to make this happen. I apologize in advance, but this is going to be lengthy.

Dependencies:
  * Using mysql instead of sqlite3
    * I've done some preliminary work on making sure that we can use mysql instead of sqlite3 for the database engine...at this point not much more than proof of concept. 
    * I need to figure out the steps to make this happen on pythonanywhere.
  * Modify stage_history.py to have “steps”
    * stage_history(step=’createweek’)
      * This will create a new week and games by importing from the main spreadsheet.
      * This will NOT import picks, even if they exist.
    * stage_history(step=’getpicks’, exclude=[list_of_players])
      * This will import picks for all players EXCEPT excluded players.
      * This will NOT import any game scores.
    * stage_history(step=’getscores’)
      * This will import game scores.
  * Move remote Player picks to main spreadsheet
    * Use excel_picks.py to create cut-n-paste spreadsheet.
  * Clean up the Home page
  * Clean up the Pick10 landing page

Once all these dependencies are met, I would like to start with a single remote player. That way I can create a spreadsheet pick sheet for them if needed until the process is verified.

Here is how a typical week might go
1. Commissioner enters new week games in main spreadsheet.
2. Run stage_history(step=createweek).
3. Website staff sets kickoff for each game, enters pick_deadline and insures lock_picks is clear.
4. Remote user enters picks via website.
5. I will create a pick spreadsheet for the remote user.
6. Commissioner snapshots the pick sheets and enters all picks in main spreadsheet at the pick deadline.
7. Run stage_history(step=getpicks) (are there going to be any issues related to pick_deadline?).
8. Let the pool run.
9. When all games are completed, run stage_history(step=’getscores’).
10. Verify that website results match spreadsheet results.

**Comment by jbholden** [comment]
**stage_history**
* Another step (or possibly integrate with 'getscores'?) is to get the winner for the week and set the week.winner field.
* As an alternative, you could just open the update games page and enter the final scores, and use the admin interface to set the winner

**Enter Picks confirmation?**
Do we want to email a player's picks after they submit, or after the deadline for confirmation?

**single remote player**
This seems like a good idea.  I would suggest someone new to the process (not me for example), so we can see if they have any issues figuring out what to do.

**registering users**
Do we want to try and register all user's so they can follow along in the pool each week?  I have been following along this week on my computer and it is pretty nice to have.  We could just make them public users for now (otherwise we might need some code to prevent them from entering picks?  it is pretty nice though to see your name highlighted in the results).  

**pick submit time**
5. I will create a pick spreadsheet for the remote user.
During this step, you need to record the submit time as well (you could just look at the modified field of picks, until the submit_time is implemented)

**Any issues step 7**
7. Run stage_history(step=getpicks) (are there going to be any issues related to pick_deadline?).
I don't think there will be any problems related to pick deadline.  The stage_history code doesn't check for any pick deadlines right?  The checks are done when rendering a page.

**Missing Picks**
Between steps 3 and 7, I'm not sure what will happen for the players without any picks.  This might make some pages like the overall results page inaccessible.  I still need to investigate this.

**Work**
Are you intending on implemented all these steps yourself or do you want to divide up some of the work?

**Comment by jbholden** [comment]
The missing picks comment in the previous post should not be an issue.  See #89

**Comment by jbholden** [comment]
Any update on this issue?

**Comment by blreams** [comment]
Not much to update on this, for now. I do have the http://collegefootballpick10.pythonanywhere.com site up and running. And I did add in the 2015 spreadsheet up through Wk06 picks (but no game scores).

I have not made any changes to stage_history() to add the step argument. But once I had the spreadsheet uploaded, I did run stage_history.main(2015) to get the DB updated. Then I entered the scores for all the games as a Player might, and that seemed to work. Finally, I went in through the Admin interface and locked the scores...noting that the tab for entering game scores disappeared.

A couple of observations about the pythonanywhere site. Running stage_history.main(2015) through Wk06 chewed up 72.50 seconds of the allotted 100 seconds of runtime. While that sounds bad, it isn't quite that bad. Once you go over the 100 seconds, it doesn't just cut you off. Rather it will continue to run in a degraded mode.

You might want to register yourself with that site, as a Player (note: I'm using separate username/email accounts for my Player and Admin access). Which reminds me. The site should probably email the Commissioners whenever there is a new registration (I'll create an issue for that).
